### PR TITLE
feat(prisma): [INFRA-1064] upgrade prisma and prisma client packages

### PR DESCRIPTION
**Pull request by Herraj**
## Goal

Upgrade `prisma` and `@prisma/client` packages. This is in hopes to address the recurring issue / incident where prisma fails to connect and we have to restart the `collections-api` ecs tasks.

Prisma v4 migration doc: https://www.prisma.io/docs/guides/upgrade-guides/upgrading-versions/upgrading-to-prisma-4#breaking-changes

## Todos

- [x] Deploy and test in DEV
- [x] Ran `npx prisma db pull` as suggested in migration doc against prod schema (main pushed to dev and used the dev db url) 

## Tickets

- [INFRA-1064](https://getpocket.atlassian.net/browse/INFRA-1064)

[INFRA-1064]: https://getpocket.atlassian.net/browse/INFRA-1064?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -28,7 +28,6 @@ model IABCategory {
   parentCollections Collection[]  @relation("CollectionIABParentCategory")
   childCollections  Collection[]  @relation("CollectionIABChildCategory")
 
-  @@unique([externalId])
```
Comment (Herraj) [inline prisma/schema.prisma:31]: According to the docs we should use `@@unique` for a [compound unique field.](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#unique-1) [The v4 upgrade docs also enforce](https://www.prisma.io/docs/guides/upgrade-guides/upgrading-versions/upgrading-to-prisma-4#explicit-unique-constraints-on-one-to-one-relations) the use of `@unique` directive in line where the field defined.

File: prisma/schema.prisma [hunk 2]
```diff
@@ -59,7 +58,6 @@ model CollectionStory {
   collection   Collection              @relation(fields: [collectionId], references: [id])
 
   @@unique([collectionId, url])
-  @@unique([externalId])
```
Comment (Herraj) [inline prisma/schema.prisma:62]: See above comment.

File: prisma/schema.prisma (hunk 1 above)
Comment (Herraj) [reply prisma/schema.prisma:31]: The upgrade doc also mentioned to remove the `references: []` from the `@relation("...", fields: [...], references: [id])`. 

But the prisma docs for `@relation` say other wise (`references` is required) 🙃. So I've left it as is for now and referred to shareable-list-api service which follows the same pattern for `@relation`.

Comment (Herraj) [reply prisma/schema.prisma:31]: And regarding `@@unique` comment above, shareable-list-api still has the older pattern and seems like that still works with v4 too?

Comment (nina-py) [reply prisma/schema.prisma:31]: The Shareable Lists API would have been set up on Prisma v.4 that probably pre-dated these additional schema changes? Unless we run the migration script there, too, we'll never know 🙃

**Comment by nina-py** [comment]
Just wanted to double-check that you run the scripts in the [Migration Guide](https://www.prisma.io/docs/guides/upgrade-guides/upgrading-versions/upgrading-to-prisma-4), you didn't manually update the schema? Looks good either way! 🚀

**Review by nina-py — DISMISSED** [review]

**Review by jpetto — DISMISSED** [review]
same question as nina - did you run the `npx prisma db pull` to update the schema? but otherwise looks good! thank you for getting this fix in 😄.

File: prisma/schema.prisma [hunk 3]
```diff
@@ -8,27 +8,23 @@ datasource db {
   url      = env("DATABASE_URL")
 }
 
-enum CollectionLanguage {
-  EN
-  DE
-}
-
```
Comment (Herraj) [inline prisma/schema.prisma:15]: this was re-add by the `npx prisma db pull` command at the bottom

**Comment by Herraj** [comment]
> Just wanted to double-check that you run the scripts in the [Migration Guide](https://www.prisma.io/docs/guides/upgrade-guides/upgrading-versions/upgrading-to-prisma-4), you didn't manually update the schema? Looks good either way! 🚀

Yeah so it was a mix of manual and running the script. The `@unique` changes I had to do manually!

**Review by jpetto — APPROVED** [review]
no small amount of changes from that `npx prisma db pull`! if all tests are passing and it's been pushed to dev, should be good. could manually do a couple things on dev to verify (create/edit a collection), but that may be overkill.

thanks herraj!

**Pull request merged**